[NVFuser] multithreading test + thread safe init

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #76259

1) add multithreading tests
2) make IrParser thread safe with std::call_once (previously, registerJitOperator could get called twice simultaneously and segfault)

davidberard98 committed:
[WIP][NVFuser] multithreading nvfuser test

[ghstack-poisoned]

davidberard98 committed:
Update on "[WIP][NVFuser] multithreading nvfuser test"




[ghstack-poisoned]

davidberard98 committed:
Update on "[NVFuser] multithreading test + thread safe init"


1) add multithreading tests
2) make IrParser thread safe with std::call_once (previously, registerJitOperator could get called twice simultaneously and segfault)

[ghstack-poisoned]

jjsjann123 reviewed (approved):
LGTM, thanks for the fix

davidberard98 commented:
@pytorchmergebot merge this please

davidberard98 committed:
Update on "[NVFuser] multithreading test + thread safe init"


1) add multithreading tests
2) make IrParser thread safe with std::call_once (previously, registerJitOperator could get called twice simultaneously and segfault)

[ghstack-poisoned]

davidberard98 commented:
@pytorchmergebot merge this
